lgc/patch/FragColorExport.cpp

```diff
@@ -29,17 +29,64 @@
  ***********************************************************************************************************************
  */
 #include "lgc/patch/FragColorExport.h"
+#include "lgc/BuilderBase.h"
 #include "lgc/LgcContext.h"
+#include "lgc/patch/Patch.h"
 #include "lgc/state/IntrinsDefs.h"
+#include "lgc/state/PalMetadata.h"
+#include "lgc/state/PipelineShaders.h"
 #include "lgc/state/PipelineState.h"
 #include "lgc/state/TargetInfo.h"
 #include "llvm/IR/Constants.h"
 #include "llvm/IR/Instructions.h"
+#include "llvm/IR/IntrinsicsAMDGPU.h"
 
 #define DEBUG_TYPE "lgc-frag-color-export"
 
+using namespace lgc;
 using namespace llvm;
 
+namespace {
+
+// The information needed for an export to a hardware color target.
+struct ColorValueInfo {
+  std::vector<Value *> value; // The value of each component to be exported.
+  unsigned location;          // The location that corresponds to the hardware color target.
+  bool isSigned;              // True if the values should be interpreted as signed integers.
+};
+
+// =====================================================================================================================
+// Pass to lower color export calls
+class LowerFragColorExport : public ModulePass {
+public:
+  LowerFragColorExport() : ModulePass(ID) {}
+  LowerFragColorExport(const LowerFragColorExport &) = delete;
+  LowerFragColorExport &operator=(const LowerFragColorExport &) = delete;
+
+  void getAnalysisUsage(AnalysisUsage &analysisUsage) const override {
+    analysisUsage.addRequired<PipelineStateWrapper>();
+    analysisUsage.addRequired<PipelineShaders>();
+  }
+
+  virtual bool runOnModule(Module &module) override;
+
+  static char ID; // ID of this pass
+
+private:
+  void updateFragColors(CallInst *callInst, ColorValueInfo expFragColors[]);
+  Value *getOutputValue(ArrayRef<llvm::Value *> expFragColor, unsigned int location, BuilderBase &builder);
```

Comment:
Please remove "llvm::" prefix in .cpp file. Apply the fix to all codes of this file.